src/input_device_monitor/app/app_handler.rs

```diff
@@ -3,41 +3,46 @@ use std::sync::Mutex;
 
 use nannou::event::WindowEvent;
 
-use crate::input_device_monitor::event_handler::concrete::KeyboardEventHandler;
-use crate::input_device_monitor::event_handler::concrete::MouseEventHandler;
-use crate::input_device_monitor::event_handler::AEventHandler;
-use crate::input_device_monitor::sender::concrete::SocketClientSender;
-use crate::input_device_monitor::sender::concrete::SOCKET_SERVER_PATH;
-use crate::input_device_monitor::sender::IEventSender;
-
-pub struct AppHandler {
-    pub event_handlers: Vec<Box<dyn AEventHandler>>,
+use crate::input_device_monitor::event_caster::concrete::flatbuffer_caster::flatbuffer_event_serializer::FlatBufferEventSerializer;
+use crate::input_device_monitor::event_handler::abstractions::AEventHandler;
+use crate::input_device_monitor::event_handler::abstractions::KeyboardEventHandler;
+use crate::input_device_monitor::event_handler::abstractions::MouseEventHandler;
+use crate::input_device_monitor::sender::abstractions::IEventSender;
+use crate::input_device_monitor::sender::concrete::sockets::SocketClientSender;
+use crate::input_device_monitor::sender::concrete::sockets::SOCKET_SERVER_PATH;
+
+pub struct AppHandler<T> {
+    pub event_handlers: Vec<Box<dyn AEventHandler<T>>>,
 }
 
-impl AppHandler {
-    pub fn new() -> Self {
-        Self {
-            event_handlers: Vec::new(),
-        }
-    }
-
-    pub fn init(&mut self) {
-        let sender: Arc<Mutex<dyn IEventSender + Send>> = Arc::new(Mutex::new(
-            SocketClientSender::new(SOCKET_SERVER_PATH).unwrap(),
-        ));
-
-        let keyboard_handler: Box<dyn AEventHandler> =
-            Box::new(KeyboardEventHandler::new(sender.clone()));
-        let mouse_handler: Box<dyn AEventHandler> =
-            Box::new(MouseEventHandler::new(sender.clone()));
-
-        self.event_handlers.push(keyboard_handler);
-        self.event_handlers.push(mouse_handler);
+impl<T> AppHandler<T> {
+    pub fn new(event_handlers: Vec<Box<dyn AEventHandler<T>>>) -> Self {
+        Self { event_handlers }
     }
 
-    pub fn handle_window_event(&self, event: &WindowEvent) {
+    pub fn handle_window_event(&self, event: &T) {
         for handler in &self.event_handlers {
             handler.handle_event(event);
         }
     }
 }
+
+pub fn init_event_window_app_handler() -> AppHandler<WindowEvent> {
+    let serilizer = Box::new(FlatBufferEventSerializer);
+    let sender: Arc<Mutex<dyn IEventSender<WindowEvent> + Send>> = Arc::new(Mutex::new(
+        SocketClientSender::new(SOCKET_SERVER_PATH, serilizer).unwrap(),
+    ));
+
+    let keyboard_handler: Box<dyn AEventHandler<WindowEvent>> =
+        Box::new(KeyboardEventHandler::new(sender.clone()));
+    let mouse_handler: Box<dyn AEventHandler<WindowEvent>> =
+        Box::new(MouseEventHandler::new(sender.clone()));
```

Comment:
`KeyboardEventHandler` and `MouseEventHandler` are really [similar](https://github.com/Pending-Name-21/screen/assets/64939916/7f4fa2fa-e081-4658-a683-035565ef6b6d). 

What about making `AEventHandler` a concrete class?
![image](https://github.com/Pending-Name-21/screen/assets/64939916/b5575a0a-e9a1-4ae7-831a-434d89d8f597)

Reply:
In the first design of the UML it was decided to make it independent to have more control over the events that we want to handle and those that we don't, separating it in this way increases the complexity but helps to keep the responsibilities separate. For example, if we no longer want to use mouse or keyboard events, we simply remove that dependency from the program, or vice versa if we want to use that event handler again, and that's it, we don't have to touch the already implemented logic.